common/persistence/sql/sqlHistoryManager.go

```diff
@@ -131,7 +132,7 @@ func (m *sqlHistoryV2Manager) AppendHistoryNodes(
 			if err != nil {
 				return err
 			}
-			if rowsAffected != 1 {
+			if !(rowsAffected == 1 || rowsAffected == 2) {
 				return fmt.Errorf("expected 1 row to be affected for tree table, got %v", rowsAffected)
```

Comment:
[Nit] is this error message slightly misleading given that 2 rows could be affected now?